File: 1389-create-target-array-in-the-given-order/1389-create-target-array-in-the-given-order.java
class Solution {
    public int[] createTargetArray(int[] nums, int[] index) {
        ArrayList<Integer> list = new ArrayList<>();
        int[] result = new int[nums.length];
        for(int i=0;i<nums.length;i++){
            list.add(index[i],nums[i]);
        }
        for(int i=0;i<list.size();i++){
            result[i] = list.get(i);
        }
        return result;
    }
}

File: 1389-create-target-array-in-the-given-order/1389-create-target-array-in-the-given-order.js
/**
 * @param {number[]} nums
 * @param {number[]} index
 * @return {number[]}
 */
var createTargetArray = function(nums, index) {
    const arr = [];
    for(let i=0;i<nums.length;i++){
        arr.splice(index[i], 0, nums[i]);
    }
    return arr;
};